### DISCUSSION TITLE
WLJS does not connect with Wolfram Engine 14

### DISCUSSION THREAD

**Discussion by amindanial** [WLJSTeam/wljs-notebook#320] [General]
Ungotunately, WLJS does not connect with Wolfram Engine 14 after starting a new notebook while Jupyter does. What could be the problem?
![Screenshot 1](https://github.com/user-attachments/assets/98a51dc1-4d48-48ee-b834-2dfd4b6b9d2e)
![Screenshot 2](https://github.com/user-attachments/assets/5d31cf39-3b1d-4c7d-b252-49e0fd1ac045)

**Comment by JerryI** [comment]
Does it throw a `Timeout` error? 
I don't know how many kernels Jupyter uses, it could be just 1, while WLJS requires 2. The license limitation is exactly 2.

It might help you, if you try to restart WLJS App with no internet connection (i've seen it on some Windows machines). Then, you can trick WE it won't recognize the number running processes or something like that. I believe, I've seen it a long time ago in the discussion on VSCode Notebook extension.

**Reply by amindanial** [reply]
Yes, it throws a Timeout Error.

**Reply by JerryI** [reply]
Please try to restart with no internet connection (disable an ethernet adapter). If it works. Then we can try to figure out why... and hot to fix it

**Comment by amindanial** [comment]
I installed engine 14.1 and everything is working fine.

Regards
Dr. Amin Danial
________________________________

**Reply by JerryI** [reply]
> I installed engine 14.1 and everything is working fine.
> 
> 
> 
> Regards
> 
> Dr. Amin Danial
> 
> ________________________________
> 
> From: Kirill Vasin ***@***.***>
> 
> Sent: Friday, December 20, 2024 10:26:18 PM
> 
> To: JerryI/wolfram-js-frontend ***@***.***>
> 
> Cc: amindanial ***@***.***>; Author ***@***.***>
> 
> Subject: Re: [JerryI/wolfram-js-frontend] WLJS does not connect with Wolfram Engine 14 (Discussion #320)
> 
> 
> 
> 
> 
> Please try to restart with no internet connection (disable an ethernet adapter). If it works. Then we can try to figure out why... and hot to fix it
> 
> 
> 
> —
> 
> Reply to this email directly, view it on GitHub<https://github.com/JerryI/wolfram-js-frontend/discussions/320#discussioncomment-11633411>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AKA6J72OFW4RXREMQ7H2SA32GR4OVAVCNFSM6AAAAABT66SRQOVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNRTGM2DCMI>.
> 
> You are receiving this because you authored the thread.Message ID: ***@***.***>
> 
> 

This is great!

However it still puzzles me, why it couldn't start the second kernel. It might be a bug of the 14.0 version.

Best,
Kirill
